dspy/teleprompt/bootstrap_finetune.py

```diff
@@ -0,0 +1,261 @@
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import dspy
+from dspy import LM  # TODO: Remove after the old LM class is removed
+from dspy.adapters.base import Adapter
+from dspy.clients.utils_finetune import infer_data_format
+from dspy.evaluate.evaluate import Evaluate
+from dspy.primitives.example import Example
+from dspy.predict.predict import Predict
+from dspy.primitives.program import Program
+from dspy.teleprompt.teleprompt import Teleprompter
+from dspy.utils.logging import logger
+
+
+class BootstrapFinetune(Teleprompter):
+
+    # TODO(PR) check with team
```

Comment:
Let's add docstring to explain what these args do (they are not very self-explanable)

Reply:
Sounds great! Holding off for now in case we decide to change the, making a note for my next update